org.jacoco.core.test/src/org/jacoco/core/test/filter/targets/Finally.java

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2017 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.test.filter.targets;
+
+import static org.jacoco.core.test.validation.targets.Stubs.ex;
+import static org.jacoco.core.test.validation.targets.Stubs.f;
+import static org.jacoco.core.test.validation.targets.Stubs.nop;
+
+public class Finally {
+
+	/**
+	 * <pre>
+	 *   InputStream in = null;
+	 *   try {
+	 *     in = ...;
+	 *     ...
+	 *   } finally {
+	 *     if (in != null) {
+	 *       in.close();
+	 *     }
+	 *   }
+	 * </pre>
+	 */
+	private static void example(boolean t) {
+		Object in = null;
+		try {
+			in = open(t);
+		} finally { // $line-example.0$
+			if (in != null) { // $line-example.1$
+				nop(); // $line-example.2$
+			} // $line-example.3$
+		} // $line-example.4$
+	}
+
+	private static Object open(boolean t) {
+		ex(t);
+		return new Object();
+	}
+
+	private static void breakStatement() {
+		for (int i = 0; i < 1; i++) { // $line-breakStatement.for$
+			try {
+				if (f()) {
+					break; // $line-breakStatement$
+				}
+			} finally {
+				nop("finally"); // $line-breakStatement.1$
+			} // $line-breakStatement.2$
+		}
+	}
+
+	private static void catchNotExecuted() {
+		try {
+			nop("try");
+		} catch (Exception e) { // $line-catchNotExecuted$
+			nop("catch");
```

Comment:
@Godin Add assertion that this line is `NOT_COVERED`? To ensure that no finally code sneaks in here.

Reply:
@marchof done

Reply:
@Godin Thanks! :white_check_mark: